From 81ac2aa77b7dd23a85635648361075a14202c053 Mon Sep 17 00:00:00 2001
From: Amir Vadai <avadai@redhat.com>
Date: Wed, 3 Dec 2014 11:11:47 -0500
Subject: [ethernet] mlx4: mlx4_en_set_settings() always fails when autoneg is set

Message-id: <3a0e4d0227d684c6a0a9ac5fb5b90e6851c00a2f.1417605033.git.avadai@redhat.com>
Patchwork-id: 101104
O-Subject: [Patch RHEL7.1] net/mlx4_en: mlx4_en_set_settings() always fails when autoneg is set
Bugzilla: 1170129
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: John Greene <jogreene@redhat.com>

From: Saeed Mahameed <saeedm@mellanox.com>

Fix ethtool set settings to not check AUTONEG_ENABLE

mlx4_en_set_settings should not check if cmd->autoneg == AUTONEG_ENABLE,
cmd->autoneg can be enabled by default and this check will fail other settings requests.
mlx4_en driver doesn't support changing autoneg value, but shouldn't fail the request
in case cmd->autoneg was set.

Fixes: d48b3ab ("net/mlx4_en: Use PTYS register to set ethtool settings (Speed)")
Signed-off-by: Saeed Mahameed <saeedm@mellanox.com>
Signed-off-by: Amir Vadai <amirv@mellanox.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
(cherry picked from commit 312df74c712252017f0c2c12289f00b146ca72b3)

Bugzilla: 1170129
Brew: 8287888
Tested at Mellanox lab.

Signed-off-by: Amir Vadai <avadai@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c b/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
index 6502bcc..12606f2 100644
--- a/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
+++ b/drivers/net/ethernet/mellanox/mlx4/en_ethtool.c
@@ -755,7 +755,7 @@ static int mlx4_en_set_settings(struct net_device *dev, struct ethtool_cmd *cmd)
         speed, cmd->advertising, cmd->autoneg, cmd->duplex);
 
  if (!(priv->mdev->dev->caps.flags2 & MLX4_DEV_CAP_FLAG2_ETH_PROT_CTRL) ||
-     (cmd->autoneg == AUTONEG_ENABLE) || (cmd->duplex == DUPLEX_HALF))
+     (cmd->duplex == DUPLEX_HALF))
   return -EINVAL;
 
  memset(&ptys_reg, 0, sizeof(ptys_reg));
-- 
1.7.1